feat: pass `docs_map` to Executor (#5366)
jina-ai/jina: docs/fundamentals/flow/executor-args.md

| `py_modules` | The customized python modules need to be imported before loading the executor<br><br>Note that the recommended way is to only import a single module - a simple python file, if your<br>executor can be defined in a single file, or an ``__init__.py`` file if you have multiple files,<br>which should be structured as a python package. For more details, please see the<br>`Executor cookbook <https://docs.jina.ai/fundamentals/executor/executor-files/>`__ | `array` | `None` |
| `output_array_type` | The type of array `tensor` and `embedding` will be serialized to.<br><br>Supports the same types as `docarray.to_protobuf(.., ndarray_type=...)`, which can be found <br>`here <https://docarray.jina.ai/fundamentals/document/serialization/#from-to-protobuf>`.<br>Defaults to retaining whatever type is returned by the Executor. | `string` | `None` |
| `exit_on_exceptions` | List of exceptions that will cause the Executor to shut down. | `array` | `[]` |

> | `disable_reduce` | Disable the built-in reduce mechanism, set this if the reduction is to be handled by the Executor, if the Executor is expected to handle `docs_matrix` | `boolean` | `False` |

Reviewer: ```suggestion
| `disable_reduce` | Disable the built-in reduction mechanism. Set this if the reduction is to be handled by the Executor itself by operating on a `docs_matrix` | `boolean` | `False` |
```

> | `docs_by_executor` | If enabled, in combination with `--disable-reduce`,the `docs_matrix` received by Executor will be a dictionary with previous Executor name in the key and docs as values | `boolean` | `False` |

Reviewer: ```suggestion
| `docs_by_executor` | If enabled, in combination with `--disable-reduce` ,the `docs_matrix` received by Executor will be a dictionary with previous Executor names as keys and DocumentArrays as values | `boolean` | `False` |
```
